File: frontend/src/types/feed.ts
export type FeedType = 'api' | 'file' | 'scraper'
export type FeedRunStatus = 'running' | 'success' | 'failure'

export interface FeedRun {
  id: string
  started_at: string
  completed_at: string | null
  status: FeedRunStatus
  observables_ingested: number
  observables_new: number
  error_message: string | null
}

export interface Feed {
  id: string
  name: string
  description: string | null
  feed_type: FeedType
  url: string | null
  config: Record<string, unknown> | null
  schedule_cron: string | null
  enabled: boolean
  is_preconfigured: boolean
  default_confidence: number
  last_run_at: string | null
  observable_count: number
  latest_run: FeedRun | null
  created_at: string
  updated_at: string
}

export interface FeedCreate {
  name: string
  description?: string
  feed_type: FeedType
  url?: string
  config?: Record<string, unknown>
  schedule_cron?: string
  enabled?: boolean
  auth_config?: Record<string, unknown>
  default_confidence?: number
}

export interface FeedUpdate {
  name?: string
  description?: string
  feed_type?: FeedType
  url?: string
  config?: Record<string, unknown>
  schedule_cron?: string
  enabled?: boolean
  auth_config?: Record<string, unknown>
  default_confidence?: number
}

File: src/cti/schemas/feed.py
import uuid
from datetime import datetime
from pydantic import BaseModel, Field
from cti.models.feed import FeedType, FeedRunStatus


class FeedRunResponse(BaseModel):
    id: uuid.UUID
    started_at: datetime
    completed_at: datetime | None
    status: FeedRunStatus
    observables_ingested: int
    observables_new: int
    error_message: str | None

    model_config = {"from_attributes": True}


class FeedResponse(BaseModel):
    id: uuid.UUID
    name: str
    description: str | None
    feed_type: FeedType
    url: str | None
    config: dict | None
    schedule_cron: str | None
    enabled: bool
    is_preconfigured: bool
    default_confidence: int
    last_run_at: datetime | None
    observable_count: int = 0
    latest_run: FeedRunResponse | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class FeedCreate(BaseModel):
    name: str = Field(max_length=256)
    description: str | None = None
    feed_type: FeedType
    url: str | None = None
    config: dict | None = None
    schedule_cron: str | None = None
    enabled: bool = True
    auth_config: dict | None = None  # Will be encrypted before storage
    default_confidence: int = Field(default=50, ge=0, le=100)


class FeedUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    feed_type: FeedType | None = None
    url: str | None = None
    config: dict | None = None
    schedule_cron: str | None = None
    enabled: bool | None = None
    auth_config: dict | None = None
    default_confidence: int | None = Field(default=None, ge=0, le=100)
